lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3824,6 +3843,47 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
     }
     return Op;
   }
+  case Intrinsic::masked_gather: {
+    if (BM->isAllowedToUseExtension(
```

Comment:
Why do we have to check this again? Is there a possibility that we would not have called transType for FixedVectorType before this?

Reply:
That's unlikely, but it will explicitly say to the one, who is ready this code next, that these instructions are only supported as a part of the appropriate extension. For the sake of performance I can rewrite these line in the following way:
if (notSupported)
   error_out
   return
do_translation

Reply:
I agree with the argument that it makes the code readable.